hack/dockerfiles/lint.Dockerfile

```diff
@@ -18,8 +18,10 @@ WORKDIR /go/src/github.com/moby/buildkit
 FROM base as golangci-lint
 ARG BUILDTAGS
 RUN --mount=target=/go/src/github.com/moby/buildkit --mount=target=/root/.cache,type=cache,sharing=locked \
-  GOARCH=amd64 golangci-lint run --build-tags "${BUILDTAGS}" && \
-  GOARCH=arm64 golangci-lint run --build-tags "${BUILDTAGS}" && \
+  GOOS=linux GOARCH=amd64 golangci-lint run --build-tags "${BUILDTAGS}" && \
```

Comment:
follow-up: maybe we need some configuration options now to enable both full matrix and current architecture. In CI we should run all but in dev I don't really want to wait 4x longer. Maybe some parallelization pattern as well instead of `&&`